drivers/flash/flash_stm32g4x.c

```diff
@@ -119,23 +113,24 @@ static int erase_page(const struct device *dev, unsigned int page)
 		return rc;
 	}
 
-#ifdef FLASH_OPTR_DBANK
-	if (page > 127) {
-		if (!(regs->OPTR & FLASH_OPTR_DBANK)) {
-			LOG_ERR("Page %d does not exist when DBANK=0", page);
-			return -EINVAL;
-		}
-
-		/* The pages to be erased is in bank 2*/
+#if defined(FLASH_OPTR_DBANK)
```

Comment:
You're assuming DBANK=1 if present.
If driver doesn't handle correctly case DBANK=0, that's ok for now.
But in this case, maybe we should we generate an error if this case is detected ?

Reply:
NEw commit add for that.